sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -206,86 +209,48 @@ private Request CreateAuthRequest(MsiType msiType, string[] scopes, string clien
-                catch (Exception e) when (!(e is OperationCanceledException))
-                {
-                    return false;
+                    var success = result.AsyncWaitHandle.WaitOne(ImdsAvailableTimeoutMs);
+
+                    return success && client.Connected;
                 }
             }
+            catch
+            {
+                return false;
+            }
         }
 
-        private async Task<bool> ImdsAvailableAsync(CancellationToken cancellationToken)
+        public virtual async Task<bool> ImdsAvailableAsync(CancellationToken cancellationToken)
         {
-            // send a request without the Metadata header.  This will result in a failed request,
-            // but we're just interested in if we get a response before the timeout of 500ms
-            // if we don't get a response we assume the imds endpoint is not available
-            using (Request request = _pipeline.HttpPipeline.CreateRequest())
+            // try to create a TCP connection to the IMDS IP address. If the connection can be established
+            // we assume that IMDS is available. If connecting times out or fails to connect assume that
+            // IMDS is not available in this environment.
+            try
             {
-                request.Method = RequestMethod.Get;
-
-                request.Uri.Reset(s_imdsEndpoint);
-
-                request.Uri.AppendQuery("api-version", ImdsApiVersion);
-
-                CancellationToken imdsTimeout = new CancellationTokenSource(ImdsAvailableTimeoutMs).Token;
-
-                try
+                using (var client = new TcpClient())
                 {
-                    AzureIdentityEventSource.Singleton.ProbeImdsEndpoint(request.Uri);
-
-                    Response response = await _pipeline.HttpPipeline.SendRequestAsync(request, CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, imdsTimeout).Token).ConfigureAwait(false);
+                    var result = client.BeginConnect(s_imdsHostIp, s_imdsPort, null, null);
 
-                    AzureIdentityEventSource.Singleton.ImdsEndpointFound(request.Uri);
+                    var success = await Task.Run<bool>(() => result.AsyncWaitHandle.WaitOne(ImdsAvailableTimeoutMs), cancellationToken).ConfigureAwait(false);
 
-                    return true;
-                }
-                // we only want to handle the case when the imdsTimeout resulted in the request being canceled.
-                // this indicates that the request timed out and that imds is not available.  If the operation
-                // was user canceled we don't wan't to handle the exception so s_identityAvailable will
-                // remain unset, as we still haven't determined if the imds endpoint is available.
-                catch (OperationCanceledException) when (!cancellationToken.IsCancellationRequested)
-                {
-                    AzureIdentityEventSource.Singleton.ImdsEndpointUnavailable(request.Uri);
```

Comment:
We are not calling this event anywhere. Should it be removed(I don't think it shoudl)? Same for one above

Reply:
No, good catch I'll add this call back in.